File: android-message-forwarder/app/src/main/java/com/insurros/androidMessageForwarder/logger/LoggingMessage.java
package com.umbrella.insurance.androidMessageForwarder.logger;

import org.json.JSONException;
import org.json.JSONObject;

import lombok.Data;

@Data
public class LoggingMessage {
    private String appName;
    private String logLevel;
    private String loggingPayload;
    private String callingLoggerName;
    private String callingMethod;
    private Double longitude;
    private Double latitude;
    private Double accuracy;
    public String toJSON(){

        JSONObject jsonObject= new JSONObject();
        try {
            jsonObject.put("appName", getAppName());
            jsonObject.put("logLevel", getLogLevel());
            jsonObject.put("loggingPayload", getLoggingPayload());
            jsonObject.put("callingLoggerName", getCallingLoggerName());
            jsonObject.put("callingMethod", getCallingMethod());
            jsonObject.put("longitude", getLongitude());
            jsonObject.put("latitude", getLatitude());
            jsonObject.put("accuracy", getAccuracy());

            return jsonObject.toString();
        } catch (JSONException e) {
            // TODO Auto-generated catch block
            e.printStackTrace();
            return "";
        }

    }
}

File: umbrella-insurance-react-ui/src/models/logging/v1/LoggingMessage.ts
export class LoggingMessage {
      appName?: string;
      logLevel?: string;
      loggingPayload?: string;
      callingLoggerName?: string;
      callingMethod?: string;
      latitude?: number;
      longitude?: number;
      accuracy?: number;
      constructor(){
            let error: Error = new Error();
            if(error?.stack) {
                  this.callingMethod = error?.stack
                  ?.split('\n')[2]
                  ?.replace(/^\s+at\s+(.+?)\s.+/g, '$1' );   
            }
      }
}